compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -172,7 +172,38 @@ object Types {
         case _ =>
           false
       }
-      cls == defn.AnyClass || loop(this)
+      loop(this)
+    }
+
+    /** Returns the top type of the lattice
+     *   - XYX.Any if XYZ extends scala.Phantom and this type is upper bounded XYZ.Any
+     *   - scala.Any otherwise
+     */
+    final def topType(implicit ctx: Context): TypeRef = {
+      val lattice = phantomLatticeType
+      if (lattice.exists) lattice.select(tpnme.Any).asInstanceOf[TypeRef]
+      else defn.AnyType
+    }
+
+    /** Returns the bottom type of the lattice
+     *   - XYZ.Nothing if XYZ extends scala.Phantom and this type is upper bounded XYZ.Any
+     *   - scala.Nothing otherwise
+     */
+    final def bottomType(implicit ctx: Context): Type = {
+      val lattice = phantomLatticeType
```

Comment:
It could. The advantage of this version is that computing a bottom type that would result in `scala.Nothing` is more efficient.